DigitalLearningSolutions.Data.Tests/DataServices/SupervisorDelegateDataServiceTests.cs

```diff
@@ -0,0 +1,123 @@
+﻿namespace DigitalLearningSolutions.Data.Tests.DataServices
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Transactions;
+    using Dapper;
+    using DigitalLearningSolutions.Data.DataServices;
+    using DigitalLearningSolutions.Data.Models.Email;
+    using DigitalLearningSolutions.Data.Models.Supervisor;
+    using DigitalLearningSolutions.Data.Tests.TestHelpers;
+    using FakeItEasy;
+    using FluentAssertions;
+    using Microsoft.Data.SqlClient;
+    using Microsoft.Extensions.Logging;
+    using NUnit.Framework;
+
+    internal class SupervisorDelegateDataServiceTests
+    {
+        private readonly Guid inviteHashForFirstSupervisorDelegateRecord =
+            Guid.Parse("72e44c4d-77bd-4bed-a254-7cc27ab32927");
+
+        private SqlConnection connection = null!;
+        private SupervisorDelegateDataService supervisorDelegateDataService = null!;
+
+        [SetUp]
+        public void Setup()
+        {
+            connection = ServiceTestHelper.GetDatabaseConnection();
+            var logger = A.Fake<ILogger<SupervisorDelegateDataService>>();
+            supervisorDelegateDataService = new SupervisorDelegateDataService(connection, logger);
+        }
+
+        [Test]
+        public void GetSupervisorDelegateRecordByInviteHash_returns_correct_record()

```

Comment:
The query in `GetSupervisorDelegateRecordByInviteHash` is wrapped in a try/catch, but there is only a test that it returns the correct record. There should also be a test that it logs an error and returns null if the same `inviteHash` exists for multiple rows

Reply:
Wrote another test that it returns null. I was having toruble checking that a call to the logger was made though, got this error: `The current proxy generator can not intercept the method Microsoft.Extensions.Logging.LoggerExtensions.LogError(Microsoft.Extensions.Logging.ILogger logger, System.String message, System.Object[] args) for the following reason:
    - Extension methods can not be intercepted since they're static.`

Reply:
Ok